ScienceDirect.js

```diff
@@ -78,6 +78,15 @@ function getPDFLink(doc, onDone) {
 		return;
 	}
 	
+	// when using a partnered institution network access, construct the the PDF URL directly
+	var pdfURLdirect = doc.querySelector('link[rel="canonical"]').href;
+	if (pdfURLdirect) {
+		pdfURLdirect = pdfURLdirect + '/pdfft?isDTMRedir=true&download=true';
```

Comment:
Can you explain the query string? Does the canonical link take you to an intermediate page? If so, is the intermediate page different from the intermediate page that `citation_pdf_url` takes you to when present?

Reply:
I had noticed when clicking the  "View PDF" or "Download PDF" (sometimes this one expands into a dropdown and "Download PDF" needs to be clicked again), the pdf opens in a new tab in an online pdf reader titled Elsevier Enhanced Reader. At the top is a blue arrow-down download button. The link behind this button always follows the same pattern: the base url + the parameters. This base url appears in a `<link rel="canonical" href="...">` element on the landing page with the View or Download PDF buttons.

It is a little hacky with the string concatting, but seems to be the easiest and shortest way to a pdfURL.

Reply:
The canonical URL appears to be just another URL to the same landing page with the View or Download PDF buttons.

Reply:
Is this PR a fix or an optimization? The existing translator should pick up PDF links, including in the drop-down (though that one only works in the Connector, where we can trigger JS, and there are technical reasons why avoiding that is actually a current priority). Did you find cases where it didn't but `rel="canonical"` was available?

An optimization would be good, particularly to avoid the drop-down link, but a concern obviously would be that it would be susceptible to changes on the site (e.g., different required parameters added to the canonical URL), and we don't currently have a way of detecting if a constructed download link is going to fail and falling back to another — we can only build our best guess for a functioning download link based on the information on the page.

So, e.g., we still might want to put this below the `.PdfEmbed` block, which is at least based on what's on the page, even if we put it above the drop-down condition.

Reply:
Well, and also, you didn't answer my question about the intermediate page. Basically I'm asking if `parseIntermediatePDFPage()` works on the canonical URL. If so, I'd be inclined to just add this below the `citation_pdf_url` line and let it still use `parseIntermediatePDFPage()` rather than hard-coding the parameters, for the reasons I give above.

But I'm also just not clear on whether the canonical URL is ever actually present when `#pdfLink` and/or `citation_pdf_url` aren't and what value that URL is actually providing, separate from the hard-coded parameters.

Reply:
> Is this PR a fix or an optimization? 
An optimization, I guess. PDF retrieval fails without it for some of links.
> Did you find cases where it didn't but rel="canonical" was available? 
Yes. 

I understand want you want to do. Let me debug with a few tests.

Reply:
Commenting my code block, I locally added comments to 
```
// Some pages still have the PDF link available
	var pdfURL = attr(doc, '#pdfLink', 'href');
	Zotero.debug("If #pdfLink existed this is the pdfURL: " + pdfURL);
	if (!pdfURL) pdfURL = attr(doc, '[name="citation_pdf_url"]', 'content');
	Zotero.debug("If citation_pdf_url existed this is the pdfURL: " + pdfURL);
	if (pdfURL && pdfURL != '#') {
		parseIntermediatePDFPage(pdfURL, onDone);
		return;
	}
...
var intermediateURL = attr(doc, '.PdfEmbed > object', 'data');
	Zotero.debug("If intermediateURL existed this is the intermediateURL: " + intermediateURL);
```
and for the DOI I am using, I get the following in the debug output:
```
(3)(+0000015): Translate: Beginning translation with RIS
(3)(+0000008): Translate: Saving item
(5)(+0000001): Translate: Running handler 0 for itemDone
(3)(+0000003): Translate: If #pdfLink existed this is the pdfURL:
(3)(+0000001): Translate: If citation_pdf_url existed this is the pdfURL:
(3)(+0000001): Translate: If intermediateURL existed this is the intermediateURL:
(3)(+0000000): Translate: Refetching HTML for PDF link
```
pdfLink, citation_pdf_url, intermediateURL did not exit and parseIntermediatePDFPage is never called.

Reply:
What's the DOI?

Reply:
[https://doi.org/10.1016/S0963-8687(96)80004-7](https://doi.org/10.1016/S0963-8687(96)80004-7) (Currie & Willcocks)